apps/docs/content/docs/editor.mdx

```diff
@@ -694,6 +694,36 @@ editor.run(
 )
 ```
 
+### Hiding shapes
+
+You can pass a `isShapeHidden` predicate when configuring tldraw. This predicate will be called for each shape to determine if it should be hidden.
+
+Hidden shapes will still be present in the store, but
+
+- They will not be shown on the canvas
+- They will not be included in hit test results via `editor.getShapeAtPoint` and `editor.getShapesAtPoint`
+
+Otherwise, they will behave as normal. If you need to restrict certain interactions with hidden shapes, e.g. preventing the user from having a hidden shape selected, you need to handle that in your own code.
+
+e.g.
+
+```tsx
+<Tldraw
+	isShapeHidden={(shape) => shape.meta.hidden}
```

Comment:
is the idea here that users may want to configure how the shape is hidden differently, so could pass `shape.meta.myHiddenProperty` and then tldraw would use that to indicate if the shape is hidden?

if so that's a nice addition, I wonder if it should default to this value so it doesn't need passing, though can see how that might conflict with existing users use of a `meta.hidden` value so this might be needed to be set explicitly.

Reply:
The thinking is that the actual logic that defines when a shape should be hidden might be more complex than just a single boolean toggle. e.g. with the 'private mode' example it depends on the owner ID as well. So in the same way that React lets you define your UI as a function of your state, we let you define the 'hiddenness' of shapes as a function of your state, to make it easier to calculate correctly.

I don't think it makes sense for us to have a default implementation of this predicate since we make no prescriptions about what people put in meta properties, and it's very simple to add in userland.

Reply:
yeah that's fair, makes sense with more context!